indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -49,6 +62,36 @@ protected void setup(Context context)
     parser = config.getParser();
     granularitySpec = config.getGranularitySpec();
     reportParseExceptions = !config.isIgnoreInvalidRows();
+
+    if (config.getPathSpec() instanceof PartitionPathSpec) {
```

Comment:
instead of doing this instance of check, how about we have a new method on the pathSpec interface that is setupMapper() which contains this logic. For other path specs it is just an empty method